lib/parser_combinator/parserCombinator.mli

```diff
@@ -0,0 +1,41 @@
+open ParserM
+
+module type ParserCombinatorSig = sig 
+    include ParserMSig
+
+    (** [++] a function used for choice *)
+    val (++) : 'a parser_m -> 'a parser_m -> 'a parser_m
+
+    (** [>>] this function ensures that [p >> q] is a parser that 
+        ignores the result of parsing [p], and just parses the remaining 
+        tokens with q *)
+    val (>>) : 'a parser_m -> 'a parser_m -> 'a parser_m
+
+    (** [<<] this function ensures that [p << q] is a parser that 
+        parses using p then q, but then ignores the result from q and just uses 
+        the result from p *)
+    val (<<) : 'a parser_m -> 'a parser_m -> 'a parser_m
+
+    (** [zero] is a parser that takes an error string and returns a parser that 
+        always errors out*)
+    val zero : error:string -> 'a parser_m
+
+    (** [item] is a parser that parses one token [x] from [x::xs] and 
+        returns [Ok (x, xs)] *)
+    val item : token parser_m
+
+    (** [star p] parses using [p] zero or more times *)
+    val star : 'a parser_m -> 'a list parser_m
+
+    (** [plus p] parses using [p] one or more times *)
+    val plus : 'a parser_m -> 'a list parser_m
+
+    (** [sat p ?error] is a parser that results in an [Error] if 
+        with error message [error] if the first token does not satisfy predicate [p],
+        otherwise returns [Ok (x, xs)] where [x] is the first token parsed and 
+        [xs] are the remaining tokens *)
```

Comment:
I think these docs should be clearer about whether the tokens in xs need to satisfy `p`

Reply:
Oh I see what you mean by "returns." What a call to `Parser.parse` (or `%`) returns isn't really relevant to how you document this, i.e. don't worry about `xs`, it's not "returned" in any real sense by this parser.